### PR TITLE
[[ Bug 17834 ]] Set message box redirect ptr to nil when corresponding object is deleted

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: engine/src/mode_development.cpp [hunk 1]
```diff
@@ -731,7 +731,7 @@ bool MCModeHandleMessageBoxChanged(MCExecContext& ctxt, MCStringRef p_string)
 			else
 				t_msg_stack -> raise();
 
-			((MCField *)MCmessageboxredirect) -> settext(0, p_string, False);
+			((MCField *)MCmessageboxredirect -> Get()) -> settext(0, p_string, False);
```
Comment (runrevmark) [inline engine/src/mode_development.cpp]: 'openrect' might have called script, so MCmessageboxredirect -> Get() _might_ return nil at this point

Comment (livecodeali) [inline engine/src/mode_development.cpp]: It might also no longer be a field!

**Comment by livecodeali** [comment]
@runrevmark Fixed!

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 68b8c69

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 68b8c699470dafea7b18cb8fffa31942edce9e39

**Comment by livecode-vulcan** [comment]
😎 test success 68b8c699470dafea7b18cb8fffa31942edce9e39
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2652)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2662)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2647)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2663)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2669)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2675)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2687)

**Pull request merged**